src/common/meta/src/ddl/create_table.rs

```diff
@@ -243,14 +246,21 @@ impl CreateTableProcedure {
             }
         }
 
-        join_all(create_region_tasks)
+        self.creator.data.state = CreateTableState::CreateMetadata;
+
+        let mut results = join_all(create_region_tasks)
             .await
             .into_iter()
             .collect::<Result<Vec<_>>>()?;
```

Comment:
If we set the state before waiting for the result, we will switch to the wrong state in the next retry.